prow/clonerefs/run.go

```diff
@@ -81,7 +82,24 @@ func (o Options) Run() error {
 		go func() {
 			defer wg.Done()
 			for ref := range input {
-				output <- cloneFunc(ref, o.SrcRoot, o.GitUserName, o.GitUserEmail, o.CookiePath, env, oauthToken)
+				// Attempt to clone the repo
+				// If the cluster is autoscaling, it seems Prow can hit DNS too hard on new nodes
+				// So we retry if have a failure we can guess is due to DNS
+				// Exponential backoff is targeted for a little over 5 minutes on the last try
+				for i := 1; i < 4; i++ {
+					record := cloneFunc(ref, o.SrcRoot, o.GitUserName, o.GitUserEmail, o.CookiePath, env, oauthToken)
+					if record.Failed {
```

Comment:
Seems like the following would be simpler:
```golang
if !record.Failed {
  output <- record
  break
}
// sleep

Reply:
I need to output when it does fail under some scenarios too